docs/root/intro/arch_overview/upstream/connection_pooling.rst

```diff
@@ -86,6 +86,21 @@ back to using HTTP/2 or HTTP/1.  This path is alpha and rapidly undergoing impro
 the default behavior result in optimal latency for internet environments, so please be patient and follow along with Envoy release notes
 to stay aprised of the latest and greatest changes.
 
+Happy Eyeballs Support
+----------------------
+
+Envoy supports Happy Eyeballs, `RFC6555 <https://tools.ietf.org/html/rfc6555>`_,
+for upstream TCP connections. This behavior is configured by setting the DNS IP address resolution policy in
+:ref:`config.cluster.v3.Cluster.DnsLookupFamily <envoy_v3_api_enum_config.cluster.v3.Cluster.DnsLookupFamily>`
```

Comment:
Also seems we should mention this runtime flag `envoy.reloadable_features.allow_multiple_dns_addresses`
https://github.com/envoyproxy/envoy/blob/8f9e11247a07cc6df0ecd62689fe290f12a9013d/source/common/common/dns_utils.cc#L37

This spend some time for me to dig out why happy eye ball doesn't work for me.

Reply:
A question actually out of the scope of this PR, just curious, why we didn't add API flag to enable/disable happy eye ball here. 

Currently, it seems the happy eye ball is enabled by setting `dns_lookup_family` as `ALL`, cluster type should be `LOGICAL_DNS`, and setting the runtime flag `envoy.reloadable_features.allow_multiple_dns_addresses`. All of those setting enable the happy eye ball feature implicitly.

Reply:
runtime flag was mentioed above.  the flag will be enabled by default once we (well, Lyft) has done production testing)

Reply:
got it, thanks!